### PR TITLE
Fix TestNewManager

### LINKED ISSUE #2460: Bug: TestNewManager failed on some build Linux machines
# Expected behavior

The unit tests should be able to finish successfully on any machines if code works correctly.
We should not expect that hostname for test Linux machine was configured with some particular rules because it can be not true:
- https://www.debian.org/doc/manuals/debian-reference/ch03.en.html#_the_hostname
- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/ch-configure_host_names

# Current behavior
```
=== RUN   TestNewManager
    TestNewManager: manager_test.go:13: Expected hostname to be 'host01.mydomain.com' but it was 'host01'
--- FAIL: TestNewManager (0.00s)
```

## Steps to reproduce

1. set the hostname for Linux node as host01.mydomain.com
2. run `./build/validate.sh` script

## Context

```
# cat /etc/hostname
host01.mydomain.com
# cat /proc/sys/kernel/hostname
host01.mydomain.com
```

### PR DESCRIPTION
**Pull request by svagner**
## Please follow the guide below

# Description
Fix test `TestNewManager`
Use only full hostname for tests (see os.Hostname())

Fixes #2460

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

- [x] TestNewManager

# Checklist:

- [x] This contribution follows the project's [code of conduct](https://github.com/bosun-monitor/bosun/blob/master/CODE_OF_CONDUCT.md)
- [x] This contribution follows the project's [contributing guidelines](https://github.com/bosun-monitor/bosun/blob/master/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: host/manager_test.go [hunk 1]
```diff
@@ -19,7 +19,10 @@ func validateManager(m Manager, expectedHostname string, t *testing.T) {
 }
 
 func TestNewManager(t *testing.T) {
-	m, err := NewManager(false)
+	// we should save fqdn for NewManager since os.Hostname will return /proc/sys/kernel/hostname and it can be fqdn
+	// underhood NewManager will use same call, but if we don't save fqdn we are trying to split that name
+	// In that can this test will fail anyway
+	m, err := NewManager(true)
```
Comment (muffix) [inline host/manager_test.go]: `os.Hostname()` calls `sysctl kern.hostname` - It only corresponds to the contents of `/proc/sys/kernel/hostname` on some Linux systems and may or may not do so in the future. 

I agree that the test should pass, but I'm a bit worried that this is changing the semantics of the test. This is now not testing the splitting anymore. Should the expected host name be different instead?

**Comment by tdinucci** [comment]
Yes, thanks for the PR 👍 

I think we probably need two tests actually:

* One where `preserveFullHostName == true`
* One where `preserveFullHostName == false` but this test must then split the copy of `expectedHostname` in the test on the first `.` (which the current test isn't doing).

If you would like to update your PR to cover both of these scenarios that would be great.  Otherwise I'll be happy to do this.

File: host/manager_test.go (hunk 1 above)
Comment (svagner) [reply host/manager_test.go]: I've rewritten test to make check more predictable:)

**Comment by tdinucci** [comment]
Once you bring your branch inline with master we should be able to merge

**Comment by svagner** [comment]
Thanks @tdinucci ! Branch was updated:)

**Pull request merged**